### ISSUE TITLE
Don't retrain from scratch when the training set is not changed

### ISSUE THREAD

**Issue by asaranti** [asaranti/GNN_Counterfactuals#9]
Since retraining with the same initial GNN weights training dataset, setting of seed, and no shuffling (batches all the same) does not guarantee the same results, check if the training set is changed at all - you can do this by checking the history also. If there is no change, then recover the last trained GNN - provided that it is not deleted.

This solution will be the most time-effective thing to do also. Nevertheless, it does not prohibit the phenomenon of instability of the results on small training set changes.
